aea/protocols/base.py

```diff
@@ -231,69 +226,66 @@ def decode(self, obj: bytes) -> Message:
         return Message(json_msg)
 
 
-class Protocol(ABC):
+class Protocol(Component):
     """
     This class implements a specifications for a protocol.
 
     It includes a serializer to encode/decode a message.
     """
 
     def __init__(
-        self, protocol_id: ProtocolId, serializer: Serializer, config: ProtocolConfig
+        self, configuration: ProtocolConfig,
     ):
         """
         Initialize the protocol manager.
 
-        :param protocol_id: the protocol id.
-        :param serializer: the serializer.
-        :param config: the protocol configurations.
+        :param configuration: the protocol configurations.
         """
-        self._protocol_id = protocol_id
-        self._serializer = serializer
-        self._config = config
+        super().__init__(configuration)
+
+        self._serializer = None  # type: Optional[Serializer]
 
     @property
     def id(self) -> ProtocolId:
         """Get the name."""
-        return self._protocol_id
+        return self._configuration.public_id
 
     @property
     def serializer(self) -> Serializer:
         """Get the serializer."""
+        assert self._serializer is not None, "Serializer not initialized yet."
         return self._serializer
 
+    @serializer.setter
```

Comment:
do we actually use the serializer property anywhere?

Reply:
in `Protocol.load`